### PR TITLE
Fix typos, spelling errors, and grammar across simulator, docs, and locales

### LINKED ISSUE #6474: 🐞 Bug : Fix typos and localization errors in simulator, docs, and config files
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Describe the bug

Several source files, documentation, and locale configuration files contain typos, spelling errors that need to be corrected.

**Spelling Errors & Typos**
* `simulator/src/sequential/Keyboard.js`
  * "charecter" → "character"

* `simulator/src/hotkey_binder/view/panel.ui.js`
  * "Press Desire Key Combination..." → "Press Desired Key Combination..."
  * Comment: "htokey" → "hotkey"
  * "user prefered" → "user preferred"

* `simulator/src/hotkey_binder/model/actions.js`
  * "cals addKeys" → "calls addKeys"
  * "keycombo" → "key combo"

* `simulator/src/listeners.js` (Comments)
  * "QuicKMenu" → "QuickMenu"
  * "lenght" → "length"
  * "Listerners" → "Listeners"
  * "quichbtn" → "quickbtn"

* `docs` & `apidoc` (e.g., `apidoc/v1/index.html`, `docs/source/includes/_assignments.md`)
  * "URL Query Paramters" → "URL Query Parameters"

* `config/locales/views/circuitverse/fr.yml`
  * "CircuitVerset" → "CircuitVerse"
  * "support@CircuitVerse. rg" → "support@CircuitVerse.org"

* `config/locales/views/circuitverse/de.yml`
  * Broken mailto strings (e.g., "support@CircuitVerse senden. rg")

**Grammar Error**
* "does not exists." → "does not exist."
  * **Affected docs:** `apidoc/v1/index.html`

### Steps To Reproduce

1. Open any of the files mentioned above
2. Navigate to the relevant lines or strings
3. Observe the spelling, grammar, or translation errors

### Expected Behavior

- Code comments, UI text, documentation, and translations should be free of spelling and grammar errors
- Locale files should contain correct and meaningful translations
- Email addresses and technical terms should not be broken or malformed

### Screenshots

_No response_

### Device Information [optional]

```markdown
- OS:
- Browser:
- version:
```

### Additional context

_No response_

### Are you working on this issue?

Yes

### PR DESCRIPTION
**Pull request by himanshujays29**
Fixes #6474 

 

#### Describe the changes you have made in this PR -
This PR fixes multiple spelling, grammar, and translation errors across source code, UI text, documentation, API docs, and localization files.

### Simulator & UI
* **`simulator/src/sequential/Keyboard.js`**: Corrected spelling of `charecter` → `character`.
* **`simulator/src/hotkey_binder/view/panel.ui.js`**:
    * Updated UI text: "Press Desire Key Combination..." → "Press **Desired** Key Combination...".
    * Fixed comments: `htokey` → `hotkey`, `user prefered` → `user preferred`.
* **`simulator/src/hotkey_binder/model/actions.js`**:
    * Fixed `cals addKeys` → `calls addKeys`.
    * Fixed `keycombo` → `key combo`.
* **`simulator/src/listeners.js` (Comments)**:
    * Fixed typos: `QuicKMenu` → `QuickMenu`, `lenght` → `length`, `Listerners` → `Listeners`, `quichbtn` → `quickbtn`.

### Documentation & API
* **`apidoc/v1/index.html`** & **`docs/source/includes/_assignments.md`**:
    * Corrected "URL Query Paramters" → "URL Query **Parameters**".
* **`apidoc/v1/index.html`**:
    * Fixed grammar: "does not exists." → "does not **exist**."

### Locales (Config)
* **`config/locales/views/circuitverse/fr.yml`**:
    * Corrected `CircuitVerset` → `CircuitVerse`.
    * Fixed broken email string: `support@CircuitVerse. rg` → `support@CircuitVerse.org`.
* **`config/locales/views/circuitverse/de.yml`**:
    * Fixed broken mailto string: `support@CircuitVerse senden. rg` → `support@CircuitVerse.org`.

### Screenshots of the UI changes (If any) -

N/A
---

## Code Understanding and AI Usage

**Did you use AI assistance (ChatGPT, Claude, Copilot, etc.) to write any part of this code?**
- [x] No, I wrote all the code myself
- [ ] Yes, I used AI assistance (continue below)

**If you used AI assistance:**
- [ ] I have reviewed every single line of the AI-generated code
- [ ] I can explain the purpose and logic of each function/component I added
- [ ] I have tested edge cases and understand how the code handles them
- [ ] I have modified the AI output to follow this project's coding standards and conventions

**Explain your implementation approach:**

1. Open  the files 
2. Navigate to the relevant lines or strings
3. Observe the spelling, grammar, or translation errors

---

## Checklist before requesting a review
- [x] I have added proper PR title and linked to the issue
- [x] I have performed a self-review of my code
- [x] **I can explain the purpose of every function, class, and logic block I added**
- [x] I understand why my changes work and have tested them thoroughly
- [x] I have considered potential edge cases and how my code handles them
- [ ] If it is a core feature, I have added thorough tests
- [x] My code follows the project's style guidelines and conventions

---

Note: Please check **Allow edits from maintainers** if you would like us to assist in the PR.

## Summary by CodeRabbit

* **Bug Fixes**
  * Corrected support email addresses in legal documentation pages
  * Fixed French translation strings including product name and heading text
  * Updated user interface label text in keyboard configuration panel

* **Documentation**
  * Fixed typos in API documentation headings and descriptions
  * Corrected typographical errors in code comments and documentation blocks

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: apidoc/v1/index.html [hunk 1]
```diff
@@ -603,7 +603,7 @@ <h3 id='possible-exceptions-3'>Possible exceptions</h3>
 </thead><tbody>
 <tr>
 <td>404</td>
-<td>User associated with given <code>email</code> does not exists</td>
```
Comment (tachyons) [inline apidoc/v1/index.html:606]: This is autogenerated, remove this change

File: config/locales/views/circuitverse/de.yml [hunk 2]
```diff
@@ -122,7 +122,7 @@ de:
         description1: '5.1 Das CircuitVerse Team besitzt und behält alle Rechte an und an dem CircuitVerse Code, dem Design, Funktionalität, und Architektur von CircuitVerse, und jede Software oder Inhalte, die über CircuitVerse bereitgestellt werden (kollektiv "die CircuitVerse IP"). Wenn Sie CircuitVerse in einer Weise verwenden wollen, die durch diese Nutzungsbedingungen nicht erlaubt ist, müssen Sie sich zuerst mit dem CircuitVerse Team in Verbindung setzen. Mit Ausnahme der Rechte, die im Rahmen dieser Nutzungsbedingungen ausdrücklich gewährt werden, erhalten Sie keine Rechte an und an einer CircuitVerse IP.'
       dmca:
         main_heading: "6. Digital Millennium Copyright Act (DMCA)"
-        description1_html: "6.1 Wenn Sie ein Urheberrechtsinhaber sind und glauben, dass Inhalte auf CircuitVerse Ihre Rechte verletzen, können Sie eine E-Mail an <a href='mailto:support@CircuitVerse.org'>support@CircuitVerse senden. rg</a>.&nbsp;"
```
Comment (tachyons) [inline config/locales/views/circuitverse/de.yml:125]: This should be fixed from the crowdin

Comment (himanshujays29) [reply config/locales/views/circuitverse/de.yml:125]: @tachyons Should I remove this change from this file as well?

**Comment by himanshujays29** [comment]
@tachyons All previous changes were reverted and removed from `apidoc/v1/index.html ` and `config/locales/views/circuitverse/de.yml`

**Comment by tachyons** [comment]
@himanshujays29 No, you still haven't reverted changes from autogenerated HTML files and locale files.

**Comment by himanshujays29** [comment]
@tachyons,
Closing this PR in favor of a newer one created on the latest upstream `master`.

Most **manual code, UI text, comments, and documentation fixes** from this PR have been **re-applied cleanly** in the new PR #6497 to avoid unrelated upstream commits.

Edits to the following files were intentionally excluded in the new PR:

- apidoc/v1/index.html – auto-generated file
- config/locales/views/circuitverse/fr.yml
- config/locales/views/circuitverse/de.yml – localization files managed via Crowdin

Those issues should be addressed through their respective workflows (auto-generation and Crowdin).

Thank you for the review and feedback.
